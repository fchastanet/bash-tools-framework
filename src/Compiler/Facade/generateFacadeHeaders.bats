#!/usr/bin/env bash

# shellcheck source=src/batsHeaders.sh
source "$(cd "${BATS_TEST_DIRNAME}/../.." && pwd)/batsHeaders.sh"

# shellcheck source=/src/Compiler/Facade/generateFacadeHeaders.sh
source "${srcDir}/Compiler/Facade/generateFacadeHeaders.sh"
# shellcheck source=/src/Filters/directive.sh
source "${srcDir}/Filters/directive.sh"

function Compiler::Facade::generateFacadeHeaders::script2Facades { #@test
  run Compiler::Facade::generateFacadeHeaders "${BATS_TEST_DIRNAME}/testsData/script2Facades.sh"
  assert_success
  assert_output "$(cat "${BATS_TEST_DIRNAME}/testsData/generateFacadeHeaders.script2Facades.expected.txt")"
}

function Compiler::Facade::generateFacadeHeaders::longerScript { #@test
  run Compiler::Facade::generateFacadeHeaders "${BATS_TEST_DIRNAME}/testsData/longerScript.sh"
  assert_success
  assert_output "$(cat "${BATS_TEST_DIRNAME}/testsData/generateFacadeHeaders.longerScript.expected.txt")"
}
