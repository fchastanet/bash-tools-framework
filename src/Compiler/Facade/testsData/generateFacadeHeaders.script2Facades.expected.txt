# FACADE
# FACADE "_includes/facade2.tpl"
